libs/frontend/feature-render/src/lib/get-page-style.ts

```diff
@@ -51,10 +54,35 @@ export const getBuildPageStyle = (site: ISite, page: IPage): IRawStyleRecord =>
       breakpointStyles,
       descSortedBreakpoints.value,
       activeBreakpoint.value,
+      true,
+    )
+
+    const nonPlaceholderStyles: IRawStylesWithSource = {}
+    const placeholderStyles: IRawStylesWithSource = {}
+
+    Object.entries(flattenedStyles).forEach(([css, style]) => {
+      if (style.sourcePseudoClass === CssPseudoClass.Placeholder) {
+        placeholderStyles[css as Css] = style
+      } else {
+        nonPlaceholderStyles[css as Css] = style
+      }
+    })
+
+    const resolvedStyle = rawStyleToResolvedStyleWithSource(
+      site.context,
+      nonPlaceholderStyles,
     )
-    const resolvedStyle = rawStyleToResolvedStyleWithSource(site.context, flattenedStyles)
     pageStyle[`#${component.id}`] = resolvedStyle
 
+    if (currentPseudoClass === CssPseudoClass.Placeholder) {
+      const resolvedPlaceholderStyle = rawStyleToResolvedStyleWithSource(
+        site.context,
+        placeholderStyles,
+      )
+      pageStyle[`#${component.id}${CssPseudoClass.Placeholder}`] =
+        resolvedPlaceholderStyle
+    }
+
```

Comment:
Hi @sampullman 

For #236 I wasn't able to reproduce the bug with the same template from staging; maybe it was fixed at some point in the past 🤔. However, I noticed that partial styles (inline styles applied to some of the content in a component) will not be effected by the selected pseudo-class, which makes sense to me, but might confuse some users in the future, and some of the style-related bugs might have something to do with this. For example:

https://github.com/samatechtw/pubstudio-builder/assets/59676941/49ae3fb1-2664-4cf9-a4d5-5c4ab88928cb

In the video I set the color of the text to red first, then try to change the color from the right menu. Since the inline styles will always overwrite all other styles, users might think right menu isn't working if they forgot they've applied some partial styles to the same component before.

---

I also noticed placeholder styles are not shown in the build mode, so I made some changes here to make it visible when `::placeholder` pseudo-class is selected.
Originally I was intended to make placeholder styles visible in the builder when default pseudo-class is selected, but then I realized properties like `border`, `background-color`, etc are also available with `::placeholder`, in order to make the behavior more consistent, maybe it's better to only show placeholder styles when `::placeholder` pseudo-class is selected, just like how all other pseudo-classes do in the builder. What do you think?

https://github.com/samatechtw/pubstudio-builder/assets/59676941/48611633-4eb3-40a7-9a2e-d8379f354599

Reply:
This all makes sense, but I am still able to reproduce the original bug. I can change any property in a reusable style with the `:hover` pseudo style set, and it doesn't show in the editor.

I just tested again, and it seems like the reusable (pseudo) styles are also not applied in preview/live mode.

Reply:
Oh, sorry about that! I didn't realize the bug is about reusable style until now 😵.
I can reproduce the bug on my end now. I'll try to fix it within this PR, thank you!

Reply:
Thanks 🙏